be/src/formats/parquet/column_reader.cpp

```diff
@@ -483,6 +483,80 @@ class StructColumnReader : public ColumnReader {
+        const std::string formatted_subfield_name =
+                case_sensitive ? col_type.field_names[i] : boost::algorithm::to_lower_copy(col_type.field_names[i]);
+
+        auto it = field_name_2_pos.find(formatted_subfield_name);
+        if (it == field_name_2_pos.end()) {
+            LOG(WARNING) << "Struct subfield name: " + formatted_subfield_name + " not found.";
+            pos[i] = -1;
+            continue;
+        }
+        pos[i] = it->second;
+    }
+}
+
+void ColumnReader::get_subfield_pos_with_pruned_type(const ParquetField& field, const TypeDescriptor& col_type,
+                                                     bool case_sensitive,
+                                                     const TIcebergSchemaField* iceberg_schema_field,
+                                                     std::vector<int32>& pos,
+                                                     std::vector<const TIcebergSchemaField*>& iceberg_schema_subfield) {
+    // For Struct type with schema change, we need consider subfield not existed suitition.
+    // When Iceberg add a new struct subfield, the original parquet file do not contains newly added subfield,
+    std::unordered_map<std::string, const TIcebergSchemaField*> subfield_name_2_field_schema{};
+    for (const auto& each : iceberg_schema_field->children) {
+        std::string format_subfield_name = case_sensitive ? each.name : boost::algorithm::to_lower_copy(each.name);
+        subfield_name_2_field_schema.emplace(format_subfield_name, &each);
+    }
+
+    std::unordered_map<int32_t, size_t> field_id_2_pos{};
+    for (size_t i = 0; i < field.children.size(); i++) {
+        field_id_2_pos.emplace(field.children[i].field_id, i);
+    }
+    for (size_t i = 0; i < col_type.children.size(); i++) {
+        const auto& format_subfield_name =
+                case_sensitive ? col_type.field_names[i] : boost::algorithm::to_lower_copy(col_type.field_names[i]);
+
+        auto iceberg_it = subfield_name_2_field_schema.find(format_subfield_name);
+        if (iceberg_it == subfield_name_2_field_schema.end()) {
+            // This suitition should not be happened, means table's struct subfield not existed in iceberg schema
+            // Below code is defensive
+            DCHECK(false) << "Struct subfield name: " + format_subfield_name + " not found in iceberg schema.";
+            pos[i] = -1;
+            iceberg_schema_subfield[i] = nullptr;
+            continue;
+        }
+
+        int32_t field_id = iceberg_it->second->field_id;
+
+        auto parquet_field_it = field_id_2_pos.find(field_id);
+        if (parquet_field_it == field_id_2_pos.end()) {
+            // Means newly added struct subfield not existed in original parquet file, we put nullptr
+            // column reader in children_reader, we will append default value for this subfield later.
+            LOG(WARNING) << "Struct subfield name: " + format_subfield_name + " not found in ParquetField.";
```

Comment:
This is expected case, don't print warning logs.

Reply:
will do it